test/abstract_client.ts

```diff
@@ -3975,6 +3975,37 @@ export default function abstractTest(server, config, ports) {
 				})
 			})
 
+			it('should return an error (via callbacks) for topic it does not have access to', function _test(t, done) {
+				const client = connect({ ...config })
+
+				server.on('client', (serverClient) => {
+					serverClient.on('connect', () => {
+						serverClient.connack(connack)
+					})
+
+					serverClient.on('subscribe', (packet) => {
+						// Send an unauthorized error
+						serverClient.suback({
+							granted: [0x87],
+							messageId: packet.messageId,
+						})
+					})
+				})
+
+				client.subscribe('$SYS/#', (subErr) => {
+					client.end(true, (endErr) => {
+						if (subErr) {
+							assert.strictEqual(
+								subErr.message,
+								'Error: Subscribe error: Not authorized',
```

Comment:
```suggestion
								'Subscribe error: Not authorized',
```

Reply:
The test fails otherwise

Reply:
0kay, I was having trouble running this test locally. Thanks